src/main/java/io/r2dbc/postgresql/PostgresqlConnectionConfiguration.java

```diff
@@ -215,6 +225,22 @@ public Builder host(String host) {
             return this;
         }
 
+        /**
+         * Configure connection initialization parameters.
+         *
+         * These parameters are applied once after creating a new connection. This is useful for setting up client-specific
+         * <a href="https://www.postgresql.org/docs/current/runtime-config-client.html#RUNTIME-CONFIG-CLIENT-FORMAT">runtime parameters</a>
+         * like statement timeouts, time zones etc.
+         *
+         * @param options the options
+         * @return this {@link Builder}
+         * @throws IllegalArgumentException if {@code options} is {@code null}
+         */
+        public Builder options(Map<String, String> options) {
+            this.options = Assert.requireNonNull(options, "options must not be null");
```

Comment:
Let's add here another assertion to make sure that `options` does not contain `null` keys or `null` values. Doing so prevents subsequent potential NPE's when sending the parameters.

Reply:
Changed it.